darts/datasets/dataset_loaders.py

```diff
@@ -113,6 +121,29 @@ def _download_dataset(self):
                 "Could not download the dataset. Reason:" + e.__repr__()
             ) from None
 
+    def _download_zip_dataset(self):
+        os.makedirs(self._root_path, exist_ok=True)
+        extracted_dir = Path(DatasetLoader._DEFAULT_DIRECTORY, "temp_dir")
+        zip_path = Path(DatasetLoader._DEFAULT_DIRECTORY, "temp.zip")
+
+        try:
+            request = requests.get(self._metadata.uri)
+            with open(zip_path, "wb") as f:
+                f.write(request.content)
+            with zipfile.ZipFile(zip_path, "r") as zip_ref:
+                zip_ref.extractall(extracted_dir)
+            self._metadata.pre_process_fn(extracted_dir, self._get_path_dataset())
```

Comment:
what happens if `pre_process_fn` is `None` ?

Reply:
`pre_process_fn` is required for any non-csv file, even if it is to move the file from the directory to the datasets folder. I added a test to check for this.

Reply:
It could be nice to use `pre_process_fn` in any case (zip file or not) if it is provided, and also support the case where it is not provided (assuming the zipped CSV file is already of the right format). WDYT?

Reply:
I think it would be nice to have a use case before adding code. Do you have any in mind?

Reply:
In fact I didn't have in mind to add any code, just re-arrange things slightly so that `pre_process_fn` is being called in all execution branches (not the zip file one only). I'm proposing this because I find it potentially slightly confusing to have such a parameter `pre_process_fn` and have it silently ignored in some cases (plain CSV file). But maybe I'm missing something?

Reply:
Maybe Im missing something, or I should rename it? the goal of the function is to take in a temp direction (the unzipped folder) and move it to the darts database folder. here's what I was trying to accomplish with the electricity dataset. First, we have to extract the zip, then move the file to the `.darts/datasets` folder. But unfortunately, the dataset is not in the standard comma-separated standard, so each comma needs to be changed to a period and each semi-colon to a comma. This way, the dataset conforms to the format of all the other datasets and can reuse the load function. 

https://github.com/unit8co/darts/blob/61eba9b9ba504239764a98257e81570cda2cb600/darts/datasets/__init__.py#L423-L427

If we move it outside the zip section, that temp folder is going to be empty so I'm not sure what processing would happen
```python 
        if not self._is_already_downloaded():
            with tempfile.TemporaryFile() as tf:
                with tempfile.TemporaryDirectory() as td:
                    if self._metadata.uri.endswith(".zip"):
                        self._download_zip_dataset(tf, td)
                    else:
                        self._download_dataset()
                    self._metadata.pre_process_fn(td, self._get_path_dataset())
        self._check_dataset_integrity_or_raise()
        return self._load_from_disk(self._get_path_dataset(), self._metadata)
```

Potential solutions would be to rename it to be less confusing and to throw an exception if the arg is provided with a csv file. I also might just not understand.

Reply:
OK I see. What I had in mind was to separate *moving* the file to the right path from *pre-processing* its content (e.g. to change periods in semi-columns). This way the `pre_process_fn` could do only the pre-processing on any CSV file, and we could move the file first (using e.g. `shutil.move`). We could also easily apply `pre_process_fn` in the function `_download_dataset()` too so that it would also be applied to any CSV file even if this CSV file has not previously been zipped. However, decoupling this way also makes things slightly more complex which is perhaps unnecessary at this point. So let's leave it as you propose. Perhaps one improvement would be to rename `pre_process_fn` to `pre_process_zipped_csv_fn` to make it clear it applies only to CSVs that are zipped. Makes sense to you?

Reply:
renamed the parameter and added a warning if the param is used with a csv file.